src/parser/permit-generation-module.ts

```diff
@@ -0,0 +1,116 @@
+import { context } from "@actions/github";
+import { Value } from "@sinclair/typebox/value";
+import { createClient } from "@supabase/supabase-js";
+import {
+  Context,
+  createAdapters,
+  Database,
+  encodePermits,
+  generatePayoutPermit,
+  SupportedEvents,
+  TokenType,
+} from "@ubiquibot/permit-generation";
+import configuration from "../configuration/config-reader";
+import permitGenerationConfigurationType, {
+  PermitGenerationConfiguration,
+} from "../configuration/permit-generation-configuration";
+import { getOctokitInstance } from "../get-authentication-token";
+import { IssueActivity } from "../issue-activity";
+import { Module, Result } from "./processor";
+
+interface Payload {
+  evmNetworkId: number;
+  issueUrl: string;
+  evmPrivateEncrypted: string;
+  issue: { id: number };
+}
+
+export class PermitGenerationModule implements Module {
+  readonly _configuration: PermitGenerationConfiguration = configuration.permitGeneration;
+  readonly _supabase = createClient<Database>(process.env.SUPABASE_URL, process.env.SUPABASE_ANON_KEY);
+
+  async transform(data: Readonly<IssueActivity>, result: Result): Promise<Result> {
+    const payload: Context["payload"] & Payload = context.payload.inputs;
```

Comment:
Why `Context["payload"]` ?

Reply:
This corresponds to the payload passed through Actions, containing the data passed through the arguments. I changed the logic to use the CLI instead inside of #12 for a few reasons:
- ease of use locally
- works regardless of the running environment
- allows for argument validation

See https://github.com/ubiquibot/conversation-rewards/pull/12/files#diff-d8695c01ce6d8a8c5009864e1cafc575147114345aec56f16cb0f0231812a4c2R32